### PR TITLE
pin flann/pyflann version to 1.8.4 to fix bug

### PR DESCRIPTION
**Pull request by yuchaz**
The Flann library will not load correctly when importing megaman build with flann version 1.9.1. Therefore, we have to pin the version to 1.8.5 for both flann and pyflann to fix this bug

### CODE REVIEW

**Comment by jakevdp** [comment]
Hmm... looks like the older version of pyflann wasn't built for Python 3.6, and that's causing the CI error. I'm not certain how to address that.

**Comment by yuchaz** [comment]
I search for the available versions of `pyflann` on anaconda, seems like `pyflann 1.8.4` supports python 2.7, 3.4 and 3.5, I will switch to 1.8.4 in next commit.

```
pyflann                      1.8.4                    py27_0  conda-forge     
                             1.8.4                    py34_0  conda-forge     
                             1.8.4                    py35_0  conda-forge     
                             1.8.4                    py36_0  conda-forge     
                             1.8.5                    py27_0  conda-forge     
                          *  1.8.5                    py35_0  conda-forge     
                             1.8.5                    py36_0  conda-forge     
                             1.9.1                    py27_0  conda-forge     
                             1.9.1                    py35_0  conda-forge     
                             1.9.1                    py36_0  conda-forge     
```

**Comment by yuchaz** [comment]
@jakevdp : do you know what is the oldest numpy version we support? Using `numpy x.x` is a deprecated pattern, and we need to change it to this pattern: `numpy >= VERSION`

**Comment by jakevdp** [comment]
I think the issue is that the current build uses 3.6, and so pinning a pyflann for which 3.6 is unavailable will cause an error. The fix has to happen in pyflann, not here.

**Comment by yuchaz** [comment]
On CI's [error log](https://circleci.com/gh/conda-forge/megaman-feedstock/30), it said that:
```
The following specifications were found to be in conflict:
  - pyflann 1.8.5 -> python 3.6*
  - python 3.4.*
```
As I understood, that is because `pyflann 1.8.5` only support python 2.7/3.5/3.6, but we are using/testing on python 3.4. And thus the error comes out. Am I interpret it wrong?

**Comment by jakevdp** [comment]
Ah, yes, I think you're right. If we update the build machinery by rerendering the feedstock as part of this PR, it will remove Python 3.4 and use 3.5/3.6.

**Comment by yuchaz** [comment]
Hi @jakevdp do you know which file I should edit to update the build machinery and rerendering the feedstock? Sorry for not really clear how to do that.

**Comment by isuruf** [comment]
@conda-forge-admin, please rerender

**Comment by isuruf** [comment]
@yuchaz, can you fix the numpy issue? You'll need a rerendering again.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -18,15 +18,15 @@ requirements:
         - python
         - numpy x.x
```
Comment (isuruf) [inline recipe/meta.yaml]: Change this to `numpy 1.11`

File: recipe/meta.yaml [hunk 2]
```diff
@@ -18,15 +18,15 @@ requirements:
...
         - cython
-        - flann
+        - flann 1.8.4
         - gcc          # [unix]
     run:
         - python
         - numpy x.x
```
Comment (isuruf) [inline recipe/meta.yaml]: Change this to `numpy >= 1.11`

**Comment by yuchaz** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (yuchaz) [reply recipe/meta.yaml]: Thank you!

**Comment by jakevdp** [comment]
We shouldn't be running the full unit test suite here. Removing ``recipe/run_tests.py`` will fix that, and should also fix the CircleCI errors.

**Comment by jakevdp** [comment]
Probably needs to be rerendered again too, since it's been five months

**Comment by yuchaz** [comment]
@conda-forge-admin, please rerender

**Comment by yuchaz** [comment]
@jakevdp sorry for the late replied seems like its been a while since we tried to fix the dependency issue. Can you merge the pull request or point me to the right person that can merge it?

**Comment by jakevdp** [comment]
@conda-forge-admin, please rerender

**Pull request closed without merging**